### ISSUE TITLE
Meta.missing_field = merge...

### ISSUE THREAD

**Issue by cajones314** [octabytes/FireO#18]
When using the Meta missing_field = merge.  I noticed that the merge setting does not update the cls._meta.field_list dict.  This causes two issues: 

- `cls._get_fields()` will not include the merged fields.
- `cls.save()/cls.update()` will not save any changes on the merged fields.

Also, I see no logic to determine what the Field type should be for the merged field--is this an oversight?  Any clarification will be greatly appreciated.

**Comment by AxeemHaider** [comment]
Missing fields not effect on `save()/update()` method. 

> Missing field manage how fields behave when they are not in `Model` but in `Firestore`
[Read Doc](https://octabyte.io/FireO/meta-class/#missing-fields)

When you are getting data from **Firestore** if there is any field which is not in `Model` class then by default it create the field in Model class for this. This is what `merge` 

If it set `ignore` simply it ignore those fields which are in Firestore but not in `Model` class
`raise_error` raise and error `FieldNotFound` if any field in Firestore but not in `Model` class

This is also mention in the doc string what the `missing_field` are

> Model config what to do with fields that are comming from firestore and not in model
                possible values are (merge, ignore, raise_error) merge is default

Logic for `missing_field` is mention in [get_field_by_column_name](https://github.com/octabytes/FireO/blob/master/src/fireo/models/model_meta.py#L212-L240)

**Comment by cajones314** [comment]
@AxeemHaider help me understand the design decision to store the merged fields into the Meta.field_list?  Because you don't include them I can save or update any data associated to a merge field.

**Comment by AxeemHaider** [comment]
Can you please let me know what exactly you want to do ?

**Comment by cajones314** [comment]
@AxeemHaider thanks for asking.  I am retrofitting an existing Firestore application which has over 50 collections containing hundreds of thousands of documents (over 500mb worth).  The documents were created in typical NoSQL fashion, in that there is not any relational integrity and the schema per collection changed multiple times throughout its years of development, so there are many fields that are no longer used or were not migrated properly to the new field name.

To that end, I am endeavoring on moving the project to use gRPC and FireO to access the underlying Firestore data model.  In order to do that, the data must be cleansed. I have started creating a command line utility to walk the Firestore collection to generate (what I intended to use) FireO data Models.  That is:

Firestore Collection -> Documents -> gather heuristics on all existing Fields -> create mapping rules in text for merging / deleting fields -> generate FireO code to clean the data.  Run the tool again and autogenerate FireO data Model classes to represent the underlying Collection structure after it has been cleaned.  The final classes will of course use `missing_field = "ignore"` 

This command line tool I am creating I will gladly contribute back to the FireO community as I am sure it will help anyone who wishes to adopt FireO as their Firestore ORM.  

**Note:** the issues I recently reported are showstoppers for me using FireO.  I have no problem contributing changes to the base project, but only if the maintainers support the changes that I'm proposing.
